htmresearch/regions/Guassian2DLocationRegion.py

```diff
@@ -0,0 +1,445 @@
+          defaultValue="probabilistic",
+          count=0,
+        ),
+        learn=dict(
+          description="A boolean flag that indicates whether or not we should "
+                      "learn by associating the location with the sensory "
+                      "input",
+          dataType="Bool",
+          accessMode="ReadWrite",
+          count=1,
+          defaultValue=False
+        ),
+        seed=dict(
+          description="Seed for the random number generator",
+          accessMode="Read",
+          dataType="UInt32",
+          count=1,
+          defaultValue=42
+        )
+      ),
+      commands=dict(
+        reset=dict(description="Clear all cell activity"),
+        activateRandomLocation=dict(description="Set the location to a random "
+                                                "point"),
+      )
+    )
+    return spec
+
+  def __init__(self,
+               moduleCount,
+               cellsPerAxis,
+               scale,
+               orientation,
+               anchorInputSize,
+               activeFiringRate,
+               bumpSigma,
+               activationThreshold=10,
+               initialPermanence=0.21,
+               connectedPermanence=0.50,
+               learningThreshold=10,
+               sampleSize=20,
+               permanenceIncrement=0.1,
+               permanenceDecrement=0.0,
+               maxSynapsesPerSegment=-1,
+               bumpOverlapMethod="probabilistic",
+               learn=False,
+               seed=42,
+               **kwargs):
+    if moduleCount <= 0 or cellsPerAxis <= 0:
+      raise TypeError("Parameters moduleCount and cellsPerAxis must be > 0")
+    if moduleCount != len(scale) or moduleCount != len(orientation):
+      raise TypeError("scale and orientation arrays len must match moduleCount")
+
+    self.moduleCount = moduleCount
+    self.cellsPerAxis = cellsPerAxis
+    self.cellCount = cellsPerAxis * cellsPerAxis
+    self.scale = list(scale)
+    self.orientation = list(orientation)
+    self.anchorInputSize = anchorInputSize
+    self.activeFiringRate = activeFiringRate
+    self.bumpSigma = bumpSigma
+    self.activationThreshold = activationThreshold
+    self.initialPermanence = initialPermanence
+    self.connectedPermanence = connectedPermanence
+    self.learningThreshold = learningThreshold
+    self.sampleSize = sampleSize
+    self.permanenceIncrement = permanenceIncrement
+    self.permanenceDecrement = permanenceDecrement
+    self.maxSynapsesPerSegment = maxSynapsesPerSegment
+    self.bumpOverlapMethod = bumpOverlapMethod
+    self.learn = learn
+    self.seed = seed
+
+    self._modules = None
+
+    PyRegion.__init__(self, **kwargs)
+
+  def initialize(self):
+    """ Initialize grid cell modules """
+
+    if self._modules is None:
+      self._modules = []
+      for i in xrange(self.moduleCount):
+        self._modules.append(ThresholdedGaussian2DLocationModule(
+          self.cellsPerAxis,
```

Comment:
Should we make this call use keyword parameters in case the order changes later? I.e.cellsPerAxis = self.cellsPerAxis, etc.?